cmd/notation/plugin/install.go

```diff
@@ -0,0 +1,315 @@
+		Aliases: []string{"add"},
+		Short:   "Install plugin",
+		Long: `Install a plugin
+
+Example - Install plugin from file system:
+  notation plugin install --file wabbit-plugin-v1.0.zip
+
+Example - Install plugin from file system with user input SHA256 checksum:
+  notation plugin install --file wabbit-plugin-v1.0.zip --sha256sum 113062a462674a0e35cb5cad75a0bb2ea16e9537025531c0fd705018fcdbc17e
+
+Example - Install plugin from file system regardless if it's already installed:
+  notation plugin install --file wabbit-plugin-v1.0.zip --force
+
+Example - Install plugin from file system with .tar.gz:
+  notation plugin install --file wabbit-plugin-v1.0.tar.gz
+
+Example - Install plugin from URL, SHA256 checksum is required:
+  notation plugin install --url https://wabbit-networks.com/intaller/linux/amd64/wabbit-plugin-v1.0.tar.gz --sha256sum f8a75d9234db90069d9eb5660e5374820edf36d710bd063f4ef81e7063d3810b
+`,
+		Args: func(cmd *cobra.Command, args []string) error {
+			if len(args) == 0 {
+				switch {
+				case opts.isFile:
+					return errors.New("missing plugin file path")
+				case opts.isURL:
+					return errors.New("missing plugin URL")
+				}
+				return errors.New("missing plugin source")
+			}
+			if len(args) > 1 {
+				return fmt.Errorf("can only insall one plugin at a time, but got %v", args)
+			}
+			opts.pluginSource = args[0]
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			switch {
+			case opts.isFile:
+				opts.pluginSourceType = notationplugin.PluginSourceTypeFile
+			case opts.isURL:
+				opts.pluginSourceType = notationplugin.PluginSourceTypeURL
+			}
+			return install(cmd, opts)
+		},
+	}
+	opts.LoggingFlagOpts.ApplyFlags(command.Flags())
+	command.Flags().BoolVar(&opts.isFile, "file", false, "install plugin from a file in file system")
+	command.Flags().BoolVar(&opts.isURL, "url", false, "install plugin from an HTTPS URL. The timeout of the download HTTPS request is set to 10 minutes")
+	command.Flags().StringVar(&opts.inputChecksum, "sha256sum", "", "must match SHA256 of the plugin source, required when \"--url\" flag is set")
+	command.Flags().BoolVar(&opts.force, "force", false, "force the installation of the plugin")
+	command.MarkFlagsMutuallyExclusive("file", "url")
+	command.MarkFlagsOneRequired("file", "url")
+	return command
+}
+
+func install(command *cobra.Command, opts *pluginInstallOpts) error {
+	// set log level
+	ctx := opts.LoggingFlagOpts.InitializeLogger(command.Context())
+	// core process
+	switch opts.pluginSourceType {
+	case notationplugin.PluginSourceTypeFile:
+		if err := installPlugin(ctx, opts.pluginSource, opts.inputChecksum, opts.force); err != nil {
+			return fmt.Errorf("plugin installation failed: %w", err)
+		}
+		return nil
+	case notationplugin.PluginSourceTypeURL:
+		if opts.inputChecksum == "" {
+			return errors.New("installing from URL requires non-empty SHA256 checksum of the plugin source")
```

Comment:
```suggestion
			return errors.New("installing from URL requires non-empty SHA256 checksum of the plugin archive")
```

Reply:
This is not always true, because we also support install from a single executable file (which is not in an archive).